### PR TITLE
Voting

### PR DESCRIPTION
**Pull request by ayredkklc**

### CODE REVIEW

**Comment by pzjiang** [comment]
github trolled me

File: src/components/table.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { useState, useEffect, useContext } from "react";
 import { GameContext } from "../components/GameProvider.js";
 import apiFunctions from "../firebase/api.jsx";
+import { click } from "@testing-library/user-event/dist/click.js";
```
Comment (noahallen) [inline src/components/table.js:4]: This is unused.

File: src/components/GameProvider.js [hunk 2]
```diff
@@ -13,9 +13,13 @@ export const GameContextProvider = ({ children }) => {
...
 		{},
 	);
+	const [roundSuccess, setRoundSuccess] = useState([]);
 	const [isAdmin, setIsAdmin] = useState();
-	const [round, setRound] = useState({}); //this might not work check it out
+	const [rounds, setRound] = useState({});
 	const [gameState, setGameState] = useState();
+	const [currentRound, setCurrentRound] = useState();
+	const [currentTrial, setCurrentTrial] = useState();
+	const [isDebugFlag, setIsDebugFlag] = useState(true);
```
Comment (noahallen) [inline src/components/GameProvider.js:22]: We should commit this with false.

File: src/firebase/api.jsx [hunk 3]
```diff
@@ -324,154 +397,239 @@ async function assignRoles(roles, gameId) {
...
+			currentTrialIndex = Object.keys(currentRound).length - 1;
+			currentTrial = currentRound[currentTrialIndex];
+			return {
+				currentRoundIndex,
+				currentRound,
+				roundObjects,
+				currentTrialIndex,
+				numRounds,
+				currentTrial,
+			};
```
Comment (noahallen) [inline src/firebase/api.jsx:433]: Delete unused comments

File: src/firebase/api.jsx [hunk 4]
```diff
@@ -324,154 +397,239 @@ async function assignRoles(roles, gameId) {
...
+	//let roundPath = "/games/" + gameId + "/rounds/";
+	/*
+	let {
+		currentRoundIndex,
+		currentRound,
+		roundObjects,
+		currentTrialIndex,
+		numRounds,
+		currentTrial,
+	} = await get(child(dbref, roundPath)).then((snapshot) => {
```
Comment (noahallen) [inline src/firebase/api.jsx:418]: Delete

File: src/firebase/api.jsx [hunk 5]
```diff
@@ -324,154 +397,239 @@ async function assignRoles(roles, gameId) {
...
+					gameID +
+					"/rounds/" +
+					currentRoundIndex +
+					"/trials/" +
+					currentTrialIndex,
+			),
+			{},
+		);
+	}
+	*/
```
Comment (noahallen) [inline src/firebase/api.jsx:551]: Delete

File: src/pages/MainGamePage.jsx [hunk 6]
```diff
@@ -40,17 +72,60 @@ const MainGamePage = () => {
...
+			1,
+		);
+	};
+
 	// open the votes pop-up
 	const handleVotes = () => {
 		setShowRoleInfo(false); // Close popup
 		setShowMyRole(false); // Close "My Role" popup if open
-		setShowVotes((prevState) => !prevState); // Close "Votes" popup
+		setShowVotes(true); // Close "Votes" popup
```
Comment (noahallen) [inline src/pages/MainGamePage.jsx:128]: Comment here seems wrong

**Review by noahallen — CHANGES_REQUESTED** [review]

**Review by noahallen — APPROVED** [review]

**Pull request merged**